app/code/core/Mage/Eav/Model/Entity/Collection/Abstract.php

```diff
@@ -167,9 +167,11 @@ protected function _prepareStaticFields()
     protected function _initSelect()
     {
         $this->getSelect()->from(array('e' => $this->getEntity()->getEntityTable()));
-        if ($this->getEntity()->getTypeId()) {
+
+        if ($this->getEntity()->getEntityTable() != Mage_Eav_Model_Entity::DEFAULT_ENTITY_TABLE && $this->getEntity()->getTypeId()) {
             $this->addAttributeToFilter('entity_type_id', $this->getEntity()->getTypeId());
         }
```

Comment:
I tested both the PR and @joshua-bn's suggestion, and found that @joshua-bn produces the correct SQL statement.

```suggestion
        $this->getSelect()->from(['e' => $this->getEntity()->getEntityTable()]);

        if ($this->getEntity()->getEntityTable() === Mage_Eav_Model_Entity::DEFAULT_ENTITY_TABLE && $this->getEntity()->getTypeId()) {
            $this->addAttributeToFilter('entity_type_id', $this->getEntity()->getTypeId());
        }
```

Test code:
```php
$collection = Mage::getResourceModel('customer/customer_collection');
$collection->getSelect()->__toString();
// @joshua-bn Correct Result: SELECT `e`.* FROM `customer_entity` AS `e`
// PR Incorrect Result: SELECT `e`.* FROM `customer_entity` AS `e` WHERE (`e`.`entity_type_id` = '1')
```

Reply:
Oops, had my boolean logic backwards with the original suggestion.. Agreed that your suggestion is correct.